Support Spring Boot 3 and `jakarta`

joshuapsteele opened the pull request:
Branch and PR inspired by Issue 869: https://github.com/crnk-project/crnk-framework/issues/869

The goal is to support Spring Boot 3.

- Bump major version
- Bump Gradle from 6.9.2 to 8.3
- Transition from `javax` to `jakarta` dependencies

Commit:
Bump major version; Update javax to jakarta throughout

Commit:
Bump Gradle to 8.3

To be compatible with java version, bump gradle version. Disable some plugins for now to get building

Commit:
Version bumps

Commit:
Bump deltaspike for CDI tests

Commit:
crnk-setup-spring-boot3 and remove 1 and 2

Attempt to remove incompatible SB setup modules and update the rest of the framework to use the SB3 module

Commit:
Disable CDI tests, more version bumps

At least now more tests are running

Commit:
Various version bumps

Commit:
Fix vertx/reactive

kjthorpe18 commented:
I'm having some difficulty with the `:crnk-integration-examples:dagger-vertx-example`: 
```
Some problems were found with the configuration of task ':crnk-integration-examples:dagger-vertx-example:proguard' (type 'ProGuardTask').
  - Type 'proguard.gradle.ProGuardTask' property 'adaptclassstrings' is missing an input or output annotation.
    
    Reason: A property without annotation isn't considered during up-to-date checking.
    
    Possible solutions:
      1. Add an input or output annotation.
      2. Mark it as @Internal.
    
    For more information, please refer to https://docs.gradle.org/8.3/userguide/validation_problems.html#missing_annotation in the Gradle documentation.
```
plus a few more errors of this flavor. I'm not sure if the version of ProGuard in use is not compatible with Gradle 8.3

Commit:
Use a different ProGuard dependency

Commit:
Update mockito, equalsverifier, tests

kjthorpe18 commented on crnk-integration-examples/openliberty-microprofile-example/build.gradle:
```diff
@@ -21,7 +21,7 @@ dependencies {
 
 	implementation project(':crnk-setup:crnk-setup-rs')
-	implementation project(':crnk-setup:crnk-setup-cdi')
+	// implementation project(':crnk-setup:crnk-setup-cdi')
```
This, and other commented out cdi-related things is temporary - I was having issues with CDI and decided to focus on other parts. Welcome to any suggestions.

Commit:
Fix hibernate dependencies

kjthorpe18 commented:
```
ValidationClientModuleFactoryTest > test FAILED
    jakarta.validation.ValidationException: HV000183: Unable to initialize 'jakarta.el.ExpressionFactory'. Check that you have the EL dependencies on the classpath, or use ParameterMessageInterpolator instead
        at app//org.hibernate.validator.messageinterpolation.ResourceBundleMessageInterpolator.buildExpressionFactory(ResourceBundleMessageInterpolator.java:211)
        at app//org.hibernate.validator.messageinterpolation.ResourceBundleMessageInterpolator.<init>(ResourceBundleMessageInterpolator.java:97)
        at app//org.hibernate.validator.internal.engine.AbstractConfigurationImpl.getDefaultMessageInterpolator(AbstractConfigurationImpl.java:575)
        at app//org.hibernate.validator.internal.engine.AbstractConfigurationImpl.getDefaultMessageInterpolatorConfiguredWithClassLoader(AbstractConfigurationImpl.java:834)
        at app//org.hibernate.validator.internal.engine.AbstractConfigurationImpl.getMessageInterpolator(AbstractConfigurationImpl.java:485)
        at app//org.hibernate.validator.internal.engine.ValidatorFactoryImpl.<init>(ValidatorFactoryImpl.java:157)
        at app//org.hibernate.validator.HibernateValidator.buildValidatorFactory(HibernateValidator.java:38)
        at app//org.hibernate.validator.internal.engine.AbstractConfigurationImpl.buildValidatorFactory(AbstractConfigurationImpl.java:453)
        at app//jakarta.validation.Validation.buildDefaultValidatorFactory(Validation.java:103)
        at app//io.crnk.validation.ValidationModule.<init>(ValidationModule.java:23)
        at app//io.crnk.validation.ValidationModule.create(ValidationModule.java:36)
        at app//io.crnk.validation.ValidationModule.create(ValidationModule.java:32)
        at app//io.crnk.validation.internal.ValidationClientModuleFactory.create(ValidationClientModuleFactory.java:10)
        at app//io.crnk.validation.internal.ValidationClientModuleFactory.create(ValidationClientModuleFactory.java:6)
        at app//io.crnk.validation.internal.ValidationClientModuleFactoryTest.test(ValidationClientModuleFactoryTest.java:25)

        Caused by:
        java.lang.NoClassDefFoundError: com/sun/el/ExpressionFactoryImpl
            at org.hibernate.validator.messageinterpolation.ResourceBundleMessageInterpolator.buildExpressionFactory(ResourceBundleMessageInterpolator.java:203)
            ... 14 more

            Caused by:
            java.lang.ClassNotFoundException: com.sun.el.ExpressionFactoryImpl
                at org.hibernate.validator.messageinterpolation.ResourceBundleMessageInterpolator.buildExpressionFactory(ResourceBundleMessageInterpolator.java:203)
                ... 14 more
```
Getting this error from many tests - I tried adding `'jakarta.el:jakarta.el-api:5.0.1'` based on stackoverflow questions about the same error (but of the `javax` variety), but no change. Also open to suggestions.

kjthorpe18 commented:
```
> Task :crnk-data:crnk-data-activiti:compileTestJava
/Users/kthorpe/git/crnk-framework/crnk-data/crnk-data-activiti/src/test/java/io/crnk/data/activiti/repository/ActivitiResourceMapperTest.java:12: error: package org.activiti.engine.form does not exist
import org.activiti.engine.form.FormProperty;
                               ^
/Users/kthorpe/git/crnk-framework/crnk-data/crnk-data-activiti/src/test/java/io/crnk/data/activiti/repository/ActivitiResourceMapperTest.java:13: error: package org.activiti.engine.form does not exist
import org.activiti.engine.form.TaskFormData;
                               ^
/Users/kthorpe/git/crnk-framework/crnk-data/crnk-data-activiti/src/test/java/io/crnk/data/activiti/repository/ActivitiResourceMapperTest.java:113: error: cannot find symbol
                List<FormProperty> formProperties = new ArrayList<>();
                     ^
  symbol:   class FormProperty
  location: class ActivitiResourceMapperTest
/Users/kthorpe/git/crnk-framework/crnk-data/crnk-data-activiti/src/test/java/io/crnk/data/activiti/repository/ActivitiResourceMapperTest.java:114: error: cannot find symbol
                FormProperty formProperty = Mockito.mock(FormProperty.class);
                ^
  symbol:   class FormProperty
  location: class ActivitiResourceMapperTest
/Users/kthorpe/git/crnk-framework/crnk-data/crnk-data-activiti/src/test/java/io/crnk/data/activiti/repository/ActivitiResourceMapperTest.java:114: error: cannot find symbol
                FormProperty formProperty = Mockito.mock(FormProperty.class);
                                                         ^
  symbol:   class FormProperty
  location: class ActivitiResourceMapperTest
/Users/kthorpe/git/crnk-framework/crnk-data/crnk-data-activiti/src/test/java/io/crnk/data/activiti/repository/ActivitiResourceMapperTest.java:119: error: cannot find symbol
                TaskFormData formData = Mockito.mock(TaskFormData.class);
                ^
  symbol:   class TaskFormData
  location: class ActivitiResourceMapperTest
/Users/kthorpe/git/crnk-framework/crnk-data/crnk-data-activiti/src/test/java/io/crnk/data/activiti/repository/ActivitiResourceMapperTest.java:119: error: cannot find symbol
                TaskFormData formData = Mockito.mock(TaskFormData.class);
```
It seems that in the most recent version of Activiti, these classes were removed: https://stackoverflow.com/questions/55693148/how-to-get-form-property-in-a-user-task-in-activiti7 . I'm not famiar with Activiti or how it is used, so if anyone would like to provide some guidance on what to do here, I'd appreciate it.

Commit:
Re-enable CDI

Uncomment CDI dependencies, declare explicit dependency on crnk-setup-cdi in the modules that use it

whgibbo commented on crnk-core/build.gradle:
```diff
@@ -1,27 +1,36 @@
+    api 'net.jodah:typetools:0.6.3'
+	api 'org.slf4j:slf4j-api:2.0.9'
+	api 'com.fasterxml.jackson.module:jackson-module-jakarta-xmlbind-annotations:2.15.2'
+	api 'jakarta.validation:jakarta.validation-api:3.0.2'
```
Would it be worth either adding variables for common dependencies or moving them to the root project.

whgibbo commented on crnk-data/crnk-data-activiti/build.gradle:
```diff
@@ -1,23 +1,25 @@
+plugins {
+	id 'java'
+	id 'java-library'
+}
```
Might be worth adding back the 

sourceCompatibility = 1.17
targetCompatibility = 1.17

kjthorpe18 replied:
Added and fixed in another location where I simply had `17`

dennisvonk replied:
I would suggest that you use this notation:
```
    sourceCompatibility = JavaVersion.VERSION_17
    targetCompatibility = JavaVersion.VERSION_17
```

whgibbo commented on crnk-gen/crnk-gen-gradle/build.gradle:
```diff
@@ -16,27 +18,28 @@ gradlePlugin {
-    compile project(':crnk-gen:crnk-gen-typescript')
-    compile project(':crnk-gen:crnk-gen-asciidoc')
-    compile project(':crnk-gen:crnk-gen-openapi')
+	implementation 'com.fasterxml.jackson.module:jackson-module-jakarta-xmlbind-annotations'
```
does the version number need to be included here?

kjthorpe18 replied:
This was actually unneeded as it gets this dependency from other modules. Removed.

whgibbo commented on crnk-gen/crnk-gen-openapi/build.gradle:
```diff
@@ -1,27 +1,30 @@
+    implementation 'io.swagger.core.v3:swagger-integration:2.0.8'
+    implementation 'io.swagger.core.v3:swagger-core:2.0.8'
+    implementation 'io.swagger.core.v3:swagger-models:2.0.8'
+    implementation 'io.swagger.parser.v3:swagger-parser:2.0.8'
```
won't these need to be updated to latest versions, for java 17 compactability ?

kjthorpe18 replied:
Updated. `parser` doesn't seem to have a `-jakarta` variant

whgibbo commented on crnk-gen/crnk-gen-openapi/build.gradle:
```diff
@@ -1,27 +1,30 @@
-    compile 'io.swagger.core.v3:swagger-core:2.0.8'
-    compile 'io.swagger.core.v3:swagger-models:2.0.8'
-    compile 'io.swagger.parser.v3:swagger-parser:2.0.8'
+    implementation 'guru.nidi:graphviz-java:0.8.3'
```
does this need upgrading to the latest version ?

kjthorpe18 replied:
Updated

whgibbo commented on crnk-gen/crnk-gen-typescript/build.gradle:
```diff
@@ -1,14 +1,17 @@
-    testCompile project(':crnk-meta')
-    testCompile project(':crnk-test')
-    testCompile project(':crnk-data:crnk-data-facet')
+    testImplementation  'commons-io:commons-io:2.5'
```
seeing different version between builds, should we update all the latest and/or but the version in a variable/property

kjthorpe18 replied:
Updated to latest throughout

whgibbo commented on crnk-home/build.gradle:
```diff
@@ -1,17 +1,22 @@
-	testCompile group: 'net.javacrumbs.json-unit', name: 'json-unit-fluent', version: '1.5.3'
-	testCompile group: 'nl.jqno.equalsverifier', name: 'equalsverifier', version: '1.7.2'
-	testCompile group: 'com.jayway.jsonpath', name: 'json-path', version: '2.2.0'
+	api  group: 'junit', name: 'junit', version: '4.12'
```
Should remove the junit4 dependency and update tests to junit 5 ?

whgibbo commented on crnk-integration-examples/dropwizard-simple-example/build.gradle:
```diff
@@ -1,11 +1,14 @@
+	implementation project(':crnk-core')
 
-	compile group: 'io.dropwizard', name: 'dropwizard-core', version: '1.0.0'
+	implementation group: 'io.dropwizard', name: 'dropwizard-core', version: '1.0.0'
```
Shouldn't this be updated.

whgibbo commented on crnk-setup/crnk-setup-spring/src/main/java/io/crnk/spring/client/RestTemplateAdapter.java:
```diff
@@ -78,7 +82,7 @@ private synchronized void initImpl() {
                     HttpComponentsClientHttpRequestFactory apacheRequestFactory =
                             (HttpComponentsClientHttpRequestFactory) impl.getRequestFactory();
-                    apacheRequestFactory.setReadTimeout(networkTimeout.intValue());
+//                    apacheRequestFactory.setReadTimeout(networkTimeout.intValue());
```
Any reason for removing?

kjthorpe18 replied:
Ah forgot about this one. The method/property were deprecated:

> As of version 6.0, setting this property has no effect.
To change the socket read timeout, use SocketConfig.Builder.setSoTimeout(Timeout), supply the resulting SocketConfig to org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManagerBuilder.setDefaultSocketConfig(SocketConfig), use the resulting connection manager for org.apache.hc.client5.http.impl.classic.HttpClientBuilder.setConnectionManager(HttpClientConnectionManager), and supply the built HttpClient to HttpComponentsClientHttpRequestFactory(HttpClient).
Deprecated
as of 6.0, in favor of SocketConfig.Builder.setSoTimeout(Timeout), see above.

whgibbo commented on crnk-testkit/build.gradle:
```diff
@@ -1,11 +1,14 @@
-	compile 'org.assertj:assertj-core:3.9.1'
-	compile 'org.apache.httpcomponents:httpclient:4.5.2'
+	implementation 'org.assertj:assertj-core:3.9.1'
+	implementation 'org.apache.httpcomponents:httpclient:4.5.2'
```
Should we move upto client5.httpclient or at least the latest version of 4.5.x

kjthorpe18 replied:
This is wrapped up in the same changes as https://github.com/crnk-project/crnk-framework/pull/882#discussion_r1347473334. I think bumping up the the latest might be the right move, but haven't gotten too deep into that one yet.

whgibbo reviewed (commented):
looks good..  Will check and build and if possible will raise a PR on your branch.

Commit:
Update swagger versions

Commit:
Update commons-io version

Commit:
Fix java 17 source/target version

Commit:
Update dropwizard version and example

Commit:
Bump graphviz-java to latest version

Commit:
Remove unneeded dependency

Commit:
A few small version bumps, remove unused deps

whgibbo commented:
@jsteelepfpt   Have started some more changes which once I get the Junit 4 -> 5 changes done I'll raise a PR into your branch..
My changes are on https://github.com/whgibbo/crnk-framework/tree/CRNK-JAKARTA-WHG currently.

kjthorpe18 commented:
Next focus is Spring MVC changes. `WebMvcMetricsAutoConfiguration` seems to be gone, and `WebMvcTagsProvider`/`DefaultWebMvcTagsProvider` is marked for removal in Spring Boot 3.2. 

From what I can tell, Crnk only overrode the DefaultWebMvcTagsProvider to "enhance" the URL in MVC metrics. We may be able to do the same with `DefaultServerRequestObservationConvention`, if necessary. @remmeier was the original author of this section in `WebMvcTagsProvider`.
See: https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-Metrics-3.0

Commit:
Start MVC Observation classes

Commit:
fixup! Start MVC Observation classes

Commit:
Bring in Spring Boot deps to projects that need them

Commit:
Add equalsverifier dependency to crnk-core

Commit:
Cleanup unneeded dependencies

Commit:
Cleanup dependencies

kjthorpe18 commented:
I'm running into test failures where Jackson (JaxbAnnotationIntrospector specifically) uses `javax` instead of `jakarta`. I've tried to use the right Jackson dependencies, but it may be a deeper issue, possibly: https://github.com/FasterXML/jackson-modules-base/issues/153

Possibly related to Jersey, but I'm not sure yet

Error for example:
```
TaskResourceRepositoryTest > checkNotEqualsNotSupported FAILED
    java.lang.NoClassDefFoundError: javax/xml/bind/annotation/XmlElement
        at com.fasterxml.jackson.module.jaxb.JaxbAnnotationIntrospector.<init>(JaxbAnnotationIntrospector.java:137)
        at com.fasterxml.jackson.module.jaxb.JaxbAnnotationIntrospector.<init>(JaxbAnnotationIntrospector.java:124)
        at com.fasterxml.jackson.module.jaxb.JaxbAnnotationModule.setupModule(JaxbAnnotationModule.java:98)
        at com.fasterxml.jackson.databind.ObjectMapper.registerModule(ObjectMapper.java:879)
        at com.fasterxml.jackson.databind.ObjectMapper.registerModules(ObjectMapper.java:1081)
        at com.fasterxml.jackson.databind.ObjectMapper.findAndRegisterModules(ObjectMapper.java:1165)
        at io.crnk.core.boot.CrnkBoot.setupObjectMapper(CrnkBoot.java:274)
        at io.crnk.core.boot.CrnkBoot.bootDiscovery(CrnkBoot.java:188)
        at io.crnk.core.boot.CrnkBoot.boot(CrnkBoot.java:182)
        at io.crnk.data.activiti.repository.ActivitiTestBase.setup(ActivitiTestBase.java:45)
        at io.crnk.data.activiti.repository.TaskResourceRepositoryTest.setup(TaskResourceRepositoryTest.java:46)

        Caused by:
        java.lang.ClassNotFoundException: javax.xml.bind.annotation.XmlElement
            at com.fasterxml.jackson.module.jaxb.JaxbAnnotationIntrospector.<init>(JaxbAnnotationIntrospector.java:137)
            ... 10 more
```

kjthorpe18 commented on crnk-core/build.gradle:
```diff
@@ -8,15 +8,11 @@ dependencies {
 	api 'org.slf4j:slf4j-api:2.0.9'
 	api 'com.fasterxml.jackson.module:jackson-module-jakarta-xmlbind-annotations:2.15.2'
 	api 'jakarta.validation:jakarta.validation-api:3.0.2'
-	api 'javax.xml.bind:jaxb-api:2.4.0-b180830.0359'
```
This may be a problem, we need jackson to use the jakarta version of this

Commit:
Use jakarta variants of jackson dependencies

If excluding crnk-setup-cdi, this allows more tests to pass without javax errors

kjthorpe18 commented:
I've been excluding `crnk-setup-cdi` to work on other failures, but when including it, I get test errors when running. I can't figure out where this `javax` dependency is coming from. Any assistance here would be appreciated!

Edit: see https://stackoverflow.com/questions/77319334/java-lang-noclassdeffounderror-javax-enterprise-context-spi-contextual-when-swi/77512767#77512767, this may be because an external dependency isn't using jakarta yet.

For example:
```
CdiTransactionRunnerTest > initializationError FAILED
    java.lang.NoClassDefFoundError: javax/enterprise/context/spi/Contextual
        at java.base/java.lang.Class.getDeclaredConstructors0(Native Method)
        at java.base/java.lang.Class.privateGetDeclaredConstructors(Class.java:3373)
        at java.base/java.lang.Class.getConstructor0(Class.java:3578)
        at java.base/java.lang.Class.getConstructor(Class.java:2271)
        at org.junit.internal.builders.AnnotatedBuilder.buildRunner(AnnotatedBuilder.java:104)
        at org.junit.vintage.engine.discovery.DefensiveAllDefaultPossibilitiesBuilder$DefensiveAnnotatedBuilder.buildRunner(DefensiveAllDefaultPossibilitiesBuilder.java:114)
        at org.junit.internal.builders.AnnotatedBuilder.runnerForClass(AnnotatedBuilder.java:86)
        at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:70)
        at org.junit.internal.builders.AllDefaultPossibilitiesBuilder.runnerForClass(AllDefaultPossibilitiesBuilder.java:37)
        at org.junit.vintage.engine.discovery.DefensiveAllDefaultPossibilitiesBuilder.runnerForClass(DefensiveAllDefaultPossibilitiesBuilder.java:57)
        at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:70)
        at org.junit.vintage.engine.discovery.ClassSelectorResolver.resolveTestClass(ClassSelectorResolver.java:66)
        at org.junit.vintage.engine.discovery.ClassSelectorResolver.resolve(ClassSelectorResolver.java:47)
        at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.lambda$resolve$2(EngineDiscoveryRequestResolution.java:135)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
        at java.base/java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1602)
        at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129)
        at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:527)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:647)
        at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.resolve(EngineDiscoveryRequestResolution.java:189)
        at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.resolve(EngineDiscoveryRequestResolution.java:126)
        at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.resolveCompletely(EngineDiscoveryRequestResolution.java:92)
        at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.run(EngineDiscoveryRequestResolution.java:83)
        at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolver.resolve(EngineDiscoveryRequestResolver.java:113)
        at org.junit.vintage.engine.discovery.VintageDiscoverer.discover(VintageDiscoverer.java:42)
        at org.junit.vintage.engine.VintageTestEngine.discover(VintageTestEngine.java:64)
        at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discoverEngineRoot(EngineDiscoveryOrchestrator.java:152)
        at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discoverSafely(EngineDiscoveryOrchestrator.java:134)
        at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discover(EngineDiscoveryOrchestrator.java:108)
        at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discover(EngineDiscoveryOrchestrator.java:80)
        at org.junit.platform.launcher.core.DefaultLauncher.discover(DefaultLauncher.java:110)
        at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
        at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
        at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:118)
        at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:93)
        at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:88)
        at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:62)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
        at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
        at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
        at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
        at jdk.proxy1/jdk.proxy1.$Proxy2.stop(Unknown Source)
        at org.gradle.api.internal.tasks.testing.worker.TestWorker$3.run(TestWorker.java:193)
        at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
        at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
        at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
        at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
        at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:113)
        at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:65)
        at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
        at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)

        Caused by:
        java.lang.ClassNotFoundException: javax.enterprise.context.spi.Contextual
            at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
            at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
            at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
            ... 59 more
```

Commit:
Start updating documentation

Commit:
Replace context.destroy() with context.stop() in AbstractJpaJerseyTest

Commit:
Replace CrnkWebMvcTagsProvider and CrnkSpringActuatorMetricsAutoConfiguration with CrnkServerRequestObservationConvention and CrnkMvcObservationAutoConfiguration; Add tests

Commit:
Clean up AutoConfiguration import file

byrondaniels commented:
Hello! is this still being worked on? im open to help finish this but i dont want to duplicate work if youre working your way through this already

kjthorpe18 commented:
> Hello! is this still being worked on? im open to help finish this but i dont want to duplicate work if youre working your way through this already

@byrondaniels Hi! I haven't been able to work on this recently, so feel free to help out if you'd like!

tory-kk commented:
Could someone invite me to the repository? Would like to help.

aawaken commented:
just wanted to give :+1: for very much looking forward to see Crnk moved to Jakarta

joshuapsteele commented:
Neither @kjthorpe18 nor I have touched this in quite some time. We welcome others to join in and carry this forward!

DerekVonk commented:
> Neither @kjthorpe18 nor I have touched this in quite some time. We welcome others to join in and carry this forward!

I see it hasn't been touched in some time and also notice the certificates or ownership on the documentation website is changed/no longer working. Do any of you know what needs to be done to fix this or can we consider this orphaned?

mflorczak commented:
> > Neither @kjthorpe18 nor I have touched this in quite some time. We welcome others to join in and carry this forward!
> 
> I see it hasn't been touched in some time and also notice the certificates or ownership on the documentation website is changed/no longer working. Do any of you know what needs to be done to fix this or can we consider this orphaned?

I think nobody has access to this repository to merge PR so I think it was orphaned. Maybe we need create fork from this repo and start new product. Idea with crnk was preaty good but we have to migrate it to spring 3.

tory-kk committed:
Javax to jakarta fixes (#1)

tory-kk commented:
Our team has prepared a separate PR with build fixes for the current PR. Could anyone please review and merge it?
https://github.com/joshuapsteele/crnk-framework/pull/3

@joshuapsteele @kjthorpe18 @whgibbo

mflorczak commented:
> Our team have prepared a separate PR with build fixes for the current PR. Could anyone please review and merge it? [joshuapsteele#3](https://github.com/joshuapsteele/crnk-framework/pull/3)

I think nobody have permission to accept and merge it.

kjthorpe18 committed:
Merge pull request #3 from tory-kk/JAVAX-TO-JAKARTA

Javax to jakarta fixes

kjthorpe18 commented:
I've merged @tory-kk's fixes into this branch, but as mentioned above I think the overall project is unmaintained, and I'm not sure who will have permission to approve & merge this PR.

tory-kk commented:
@kjthorpe18 thank you :) What else remains to be done to prepare the branch for the merge and release?
@remmeier would you be able to merge and release a new version when the PR is ready?

Illapikov committed:
Remove spring-boot-starter-actuator from dependencies as it is no longer used

kjthorpe18 committed:
Merge pull request #4 from tory-kk/remove-actuator

dshannon-gsa commented:
@kjthorpe18 @remmeier anyone heard any news on a possible merge? We are looking to hopefully move up to Spring Boot 3 asap without moving away from crnk. Thanks in advance!

tory-kk committed:
Update Spring and Jetty (#3)

tory-kk committed:
Fix dependency version conflict (#4)

tory-kk commented:
@kjthorpe18 Could you please review and merge PR with dependency updates https://github.com/joshuapsteele/crnk-framework/pull/6 ?

kjthorpe18 commented:
@tory-kk I'm not a maintainer on this project and can't merge to master. I'm pretty sure nobody around here is, so I'd recommend forking the project and maintaining your own version.

tory-kk commented:
@kjthorpe18 my PR is for `joshuapsteele:JAVAX-TO-JAKARTA` branch (previously you merged our other PRs).
And yes, we have forked the repo with Spring Boot 3 and Jakarta and using this version in our project. Just wanted to sync our changes in this branch (maybe someday the project will be maintained again (*fingers crossed*)).

kjthorpe18 committed:
Merge pull request #6 from tory-kk/JAVAX-TO-JAKARTA

joshuapsteele commented:
Hi everyone, apologies, but neither myself nor @kjthorpe18 are maintainers of this crnk project. So, we can't merge these changes in. Like katharsis before it, crnk seems to have died on the vine. You're welcome to take the changes on this branch and run with them on your own, but I'm going to close this PR due to the lack of project maintainer interaction and the inability to merge it myself.
